TrainingJobRouteDecisionInfoTransportActionTests.testNodeOperation appears flaky on Windows

## Description

It seems that for TrainingJobRouteDecisionInfoTransportActionTests, the initial cluster state is not yet set for Windows. TrainingJobRouteDecisionInfoTransportActionTests uses the OpenSearchSingleNodeTestCase class, which spins up a node for the test to use: https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/test/OpenSearchSingleNodeTestCase.java#L96-L99. 

```
REPRODUCE WITH: gradlew ':test' --tests "org.opensearch.knn.plugin.transport.TrainingJobRouteDecisionInfoTransportActionTests.testNodeOperation" -Dtests.seed=97A7E44F40B3C64C -Dtests.security.manager=false -Dtests.locale=ca-ES -Dtests.timezone=America/Port-au-Prince -Druntime.java=17
org.opensearch.knn.plugin.transport.TrainingJobRouteDecisionInfoTransportActionTests > testNodeOperation FAILED
    java.lang.AssertionError: initial cluster state not set yet

        at __randomizedtesting.SeedInfo.seed([97A7E44F40B3C64C]:0)

        at org.opensearch.cluster.service.ClusterApplierService.state(ClusterApplierService.java:215)
Suite: Test class org.opensearch.knn.plugin.transport.TrainingJobRouteDecisionInfoTransportActionTests
        at org.opensearch.cluster.service.ClusterService.state(ClusterService.java:172)
        at org.opensearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.<init>(TransportSingleShardAction.java:173)
        at org.opensearch.action.support.single.shard.TransportSingleShardAction.doExecute(TransportSingleShardAction.java:125)
        at org.opensearch.action.support.single.shard.TransportSingleShardAction.doExecute(TransportSingleShardAction.java:78)
        at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:218)
        at org.opensearch.action.support.TransportAction.execute(TransportAction.java:188)
        at org.opensearch.action.support.TransportAction.execute(TransportAction.java:107)
        at org.opensearch.client.node.NodeClient.executeLocally(NodeClient.java:110)
        at org.opensearch.client.node.NodeClient.doExecute(NodeClient.java:97)
        at org.opensearch.client.support.AbstractClient.execute(AbstractClient.java:476)
        at org.opensearch.client.support.AbstractClient.execute(AbstractClient.java:463)
        at org.opensearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:66)
        at org.opensearch.knn.indices.ModelDao$OpenSearchKNNModelDao.lambda$get$9(ModelDao.java:384)
        at org.opensearch.knn.indices.ModelDao.runWithStashedThreadContext(ModelDao.java:705)
        at org.opensearch.knn.indices.ModelDao$OpenSearchKNNModelDao.get(ModelDao.java:379)
        at org.opensearch.knn.training.TrainingJobRunner.serializeModel(TrainingJobRunner.java:169)
        at org.opensearch.knn.training.TrainingJobRunner.lambda$train$4(TrainingJobRunner.java:135)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
```

Examples: 
1. https://github.com/opensearch-project/k-NN/actions/runs/7359362966/job/20036284435?pr=1364
2.

--- a/src/test/java/org/opensearch/knn/plugin/transport/TrainingJobRouteDecisionInfoTransportActionTests.java
+++ b/src/test/java/org/opensearch/knn/plugin/transport/TrainingJobRouteDecisionInfoTransportActionTests.java
@@ -11,98 +11,50 @@
 
 package org.opensearch.knn.plugin.transport;
 
-import org.junit.After;
-import org.junit.Before;
-import org.opensearch.core.action.ActionListener;
-import org.opensearch.action.index.IndexResponse;
-import org.opensearch.core.index.shard.ShardId;
-import org.opensearch.knn.KNNSingleNodeTestCase;
-import org.opensearch.knn.indices.Model;
-import org.opensearch.knn.indices.ModelDao;
-import org.opensearch.knn.indices.ModelMetadata;
-import org.opensearch.knn.indices.ModelState;
-import org.opensearch.knn.training.TrainingJob;
+import org.mockito.MockedStatic;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.knn.KNNTestCase;
 import org.opensearch.knn.training.TrainingJobRunner;
 import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
 
-import java.io.IOException;
-import java.util.concurrent.*;
+import java.util.Collections;
 
 import static org.mockito.Mockito.any;
-import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doNothing;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
 import static org.mockito.Mockito.when;
-import static org.opensearch.knn.common.KNNConstants.MODEL_INDEX_NAME;
-import static org.opensearch.knn.common.KNNConstants.TRAIN_THREAD_POOL;
 
-public class TrainingJobRouteDecisionInfoTransportActionTests extends KNNSingleNodeTestCase {
-
-    ExecutorService executorService;
-
-    @Before
-    public void setup() {
-        executorService = Executors.newSingleThreadExecutor();
-    }
-
-    @After
-    public void teardown() {
-        executorService.shutdown();
-    }
-
-    @SuppressWarnings("unchecked")
-    public void testNodeOperation() throws IOException, InterruptedException, ExecutionException {
-        // Ensure initial value of train job count is 0
-        TrainingJobRouteDecisionInfoTransportAction action = node().injector()
-            .getInstance(TrainingJobRouteDecisionInfoTransportAction.class);
-
-        TrainingJobRouteDecisionInfoNodeRequest request = new TrainingJobRouteDecisionInfoNodeRequest();
-
-        TrainingJobRouteDecisionInfoNodeResponse response1 = action.nodeOperation(request);
-        assertEquals(0, response1.getTrainingJobCount().intValue());
-
-        // Setup mocked training job
-        String modelId = "model-id";
-        Model model = mock(Model.class);
-        ModelMetadata modelMetadata = mock(ModelMetadata.class);
-        when(modelMetadata.getState()).thenReturn(ModelState.TRAINING);
-        when(model.getModelMetadata()).thenReturn(modelMetadata);
-        TrainingJob trainingJob = mock(TrainingJob.class);
-        when(trainingJob.getModelId()).thenReturn(modelId);
-        when(trainingJob.getModel()).thenReturn(model);
-        doAnswer(invocationOnMock -> null).when(trainingJob).run();
-
-        ModelDao modelDao = mock(ModelDao.class);
-        when(modelDao.get(modelId)).thenReturn(model);
-
-        // Here we check to make sure there is a running job
-        doAnswer(invocationOnMock -> {
-            TrainingJobRouteDecisionInfoNodeResponse response2 = action.nodeOperation(request);
-            assertEquals(1, response2.getTrainingJobCount().intValue());
-
-            IndexResponse indexResponse = new IndexResponse(new ShardId(MODEL_INDEX_NAME, "uuid", 0), modelId, 0, 0, 0, true);
-            ((ActionListener<IndexResponse>) invocationOnMock.getArguments()[1]).onResponse(indexResponse);
-            return null;
-        }).when(modelDao).put(any(Model.class), any(ActionListener.class));
-
-        // Set up the rest of the training logic
-        final CountDownLatch inProgressLatch = new CountDownLatch(1);
-        ActionListener<IndexResponse> responseListener = ActionListener.wrap(
-            indexResponse -> { inProgressLatch.countDown(); },
-            e -> fail("Failure should not have occurred")
-        );
-
-        doAnswer(invocationOnMock -> {
-            responseListener.onResponse(mock(IndexResponse.class));
-            return null;
-        }).when(modelDao).update(model, responseListener);
-
+public class TrainingJobRouteDecisionInfoTransportActionTests extends KNNTestCase {
+    public void testNodeOperation() {
+        // Initialize mocked variables for the class
+        DiscoveryNode node = mock(DiscoveryNode.class);
+        when(clusterService.localNode()).thenReturn(node);
         ThreadPool threadPool = mock(ThreadPool.class);
-        when(threadPool.executor(TRAIN_THREAD_POOL)).thenReturn(executorService);
+        TransportService transportService = mock(TransportService.class);
+        doNothing().when(transportService).registerRequestHandler(any(), any(), any(), any());
+        ActionFilters actionFilters = new ActionFilters(Collections.emptySet());
 
-        // Initialize runner and execute job
-        TrainingJobRunner.initialize(threadPool, modelDao);
-        TrainingJobRunner.getInstance().execute(trainingJob, responseListener);
+        TrainingJobRouteDecisionInfoTransportAction trainingJobRouteDecisionInfoTransportAction =
+            new TrainingJobRouteDecisionInfoTransportAction(threadPool, clusterService, transportService, actionFilters);
 
-        assertTrue(inProgressLatch.await(100, TimeUnit.SECONDS));
+        try (MockedStatic<TrainingJobRunner> mockedTrainingJobRunnerStatic = mockStatic(TrainingJobRunner.class)) {
+            // Ensure the job count is correct
+            int initialJobCount = 4;
+            final TrainingJobRunner mockedTrainingJobRunner = mock(TrainingJobRunner.class);
+            when(mockedTrainingJobRunner.getJobCount()).thenReturn(initialJobCount);
+            mockedTrainingJobRunnerStatic.when(TrainingJobRunner::getInstance).thenReturn(mockedTrainingJobRunner);
+
+            TrainingJobRouteDecisionInfoNodeRequest request = new TrainingJobRouteDecisionInfoNodeRequest();
+            TrainingJobRouteDecisionInfoNodeResponse response = trainingJobRouteDecisionInfoTransportAction.nodeOperation(request);
+            assertEquals(initialJobCount, response.getTrainingJobCount().intValue());
+
+            int resetJobCount = 0;
+            when(mockedTrainingJobRunner.getJobCount()).thenReturn(resetJobCount);
+            response = trainingJobRouteDecisionInfoTransportAction.nodeOperation(request);
+            assertEquals(resetJobCount, response.getTrainingJobCount().intValue());
+        }
     }
 }

